homeassistant/util/file.py

```diff
@@ -36,7 +36,7 @@ def write_utf8_file_atomic(
     try:
         with AtomicWriter(filename, overwrite=True).open() as fdesc:
             if not private:
-                os.fchmod(fdesc.fileno(), 0o644)
```

Comment:
The `chmod` method also accepts a file descriptor, why change it to the name?

Reply:
Doesn't work on windows
`TypeError: chmod: path should be string, bytes or os.PathLike, not int`

Reply:
uh...
<https://docs.python.org/3/library/os.html#os.chmod>

![image](https://user-images.githubusercontent.com/195327/146239939-a14df10c-a13e-4b88-8bfe-dfedf5f48be2.png)

![image](https://user-images.githubusercontent.com/195327/146239960-198b194b-37d5-4ea6-b768-1b2116d47e92.png)


![image](https://user-images.githubusercontent.com/195327/146240012-5b52f0b9-f130-45a2-8e51-8aa872fa889c.png)

Reply:
> This would fix error on Windows

Additionally curiosity, WSL should support this already? What did change?

Reply:
I have also checked the same page and tried that, no idea why its not working in windows python 3.9 with file descriptor.

BTW I am using Winpython64-3.9.8.0dot.exe from https://github.com/winpython/winpython as a portable test install

Reply:
> BTW I am using Winpython64-3.9.8.0dot.exe from winpython/winpython as a portable test install

We don't support running Home Assistant running directly on Windows.